pkg/scheduler/pool.go

```diff
@@ -35,6 +36,7 @@ type WorkerPoolController interface {
 	AddWorkerToMachine(cpu int64, memory int64, gpuType string, gpuCount uint32, machineId string) (*types.Worker, error)
 	Name() string
 	FreeCapacity() (*WorkerPoolCapacity, error)
+	Context() context.Context
```

Comment:
why are we exposing this? might be missing something but I don't see where it's used

Reply:
Its being used here - https://github.com/beam-cloud/beta9/pull/678/files#diff-655f521cd80c31efe08ad17bdf5340cc513ffb2710fa3e12aa845660f24c6b88R40 - in the worker pool sizer start function.

Reply:
WorkerPoolSizer takes a controller interface hence the addition of this function.